src/main/java/io/cryostat/core/net/JFRConnection.java

```diff
@@ -148,6 +161,43 @@ public synchronized int getPort() {
         }
     }
 
+    public synchronized String getJvmId() throws AttributeNotFoundException, InstanceNotFoundException, MBeanException, ReflectionException, IOException {
+        if (!isConnected()) {
+            connect();
+        }
+        List<String> attrNames =
+                new ArrayList<>(
+                        Arrays.asList(
+                                "ClassPath",
+                                "Name",
+                                "InputArguments",
+                                "LibraryPath",
+                                "VmVendor",
+                                "VmVersion",
+                                "StartTime"));
+
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(512);
+                DataOutputStream dos = new DataOutputStream(baos)) {
+            for (String attr : attrNames) {
+                Object attrObject =
+                        this.rjmxConnection.getAttributeValue(
+                                new MRI(Type.ATTRIBUTE, ConnectionToolkit.RUNTIME_BEAN_NAME, attr));
+
+                if (attrObject instanceof String || attrObject instanceof Long) {
+                    dos.writeUTF(attrObject.toString());
+                } else if (attrObject instanceof String[]) {
+                    dos.writeUTF(Arrays.toString((String[]) attrObject));
+                } else {
+                    throw new IllegalArgumentException(
+                            String.format(
+                                    "Unexpected attribute: {%s} of type: {%s}", attr, attrObject.getClass()));
```

Comment:
I think I'd rather be lenient than strict in this case. Probably better to have generated IDs that turn out to be non-unique for two URLs pointing at the same JVM than to simply refuse to generate any ID at all for such JVMs. Something like:

```
var attrO = getAttributeValue(runtimeBean, attr)
if isArray(attrO) dos.writeUTF(Arrays.toString(attrO)
else dos.writeUTF(attrO.toString())
```